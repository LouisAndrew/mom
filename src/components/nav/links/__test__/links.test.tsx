// import React from 'react'
// import ReactDOM from 'react-dom'
// import renderer from 'react-test-renderer'

// import { render, cleanup } from '@testing-library/react'
// import '@testing-library/jest-dom'

describe('Link component on navigation', () => {
    expect(true).toBe(true);
    // afterEach(cleanup)

    // it('renders without crashing', () => {

    //     const div = document.createElement('div')
    //     ReactDOM.render(, div)
    // })

    // it('renders correctly', () => {

    //     const { getByTestId } = render()
    // })

    // it('matches snapshot', () => {

    //     const tree = renderer.create().toJSON()
    //     expect(tree).toMatchSnapshot()
    // })
});
