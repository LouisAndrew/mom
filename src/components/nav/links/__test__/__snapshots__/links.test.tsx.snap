// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component on navigation matches snapshot 1`] = `
<ul
  className="sc-fzpans defOK"
  display="flex"
  height={
    Array [
      "fit-content",
      "fit-content",
      "fit-content",
    ]
  }
  overflow={
    Array [
      "hidden",
      "hidden",
      "visible",
    ]
  }
  width={
    Array [
      "100vw",
      "100vw",
      "auto",
    ]
  }
>
  <li
    className="sc-fzplWN hfowge"
    fontFamily="body"
    fontSize={
      Array [
        1,
        2,
      ]
    }
    fontWeight="heading"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M21.66 10.25l-9-8a1 1 0 0 0-1.32 0l-9 8a1 1 0 0 0-.27 1.11A1 1 0 0 0 3 12h1v9a1 1 0 0 0 1 1h14a1 1 0 0 0 1-1v-9h1a1 1 0 0 0 .93-.64a1 1 0 0 0-.27-1.11zM13 20h-2v-3a1 1 0 0 1 2 0zm5 0h-3v-3a3 3 0 0 0-6 0v3H6v-8h12zM5.63 10L12 4.34L18.37 10z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      Home
    </a>
  </li>
  <li
    className="sc-fzplWN kZpjky"
    fontFamily="body"
    fontSize={
      Array [
        1,
        2,
      ]
    }
    fontWeight="heading"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className=""
      href="/products"
      onClick={[Function]}
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M19 7h-3V6a4 4 0 0 0-8 0v1H5a1 1 0 0 0-1 1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8a1 1 0 0 0-1-1zm-9-1a2 2 0 0 1 4 0v1h-4zm8 13a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V9h2v1a1 1 0 0 0 2 0V9h4v1a1 1 0 0 0 2 0V9h2z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      Katalog
      <span
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <svg
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"currentColor\\"/>",
            }
          }
          focusable={false}
          height="1em"
          id="expand-svg"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "transform": "rotate(360deg)",
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        />
      </span>
    </a>
    <div
      className="sc-fzozJi kNjACt"
      display="flex"
      height={
        Array [
          0,
          0,
          "auto",
        ]
      }
      opacity={
        Array [
          1,
          1,
          0,
        ]
      }
      overflow="hidden"
      width={
        Array [
          1,
          1,
          0,
        ]
      }
    >
      <ul
        className="sc-AxheI eZjegr"
        display="block"
      >
        <li
          className="sc-Axmtr ghtafS"
          color="bg"
          fontFamily="body"
          fontSize={
            Array [
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          id="house"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/products?prop=house"
            onClick={[Function]}
          >
            Rumah
          </a>
        </li>
        <li
          className="sc-Axmtr ghtafS"
          color="bg"
          fontFamily="body"
          fontSize={
            Array [
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          id="apartment"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/products?prop=apartment"
            onClick={[Function]}
          >
            Apartment
          </a>
        </li>
        <li
          className="sc-Axmtr ghtafS"
          color="bg"
          fontFamily="body"
          fontSize={
            Array [
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          id="kavling"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/products?prop=kavling"
            onClick={[Function]}
          >
            Kavling
          </a>
        </li>
        <li
          className="sc-Axmtr ghtafS"
          color="bg"
          fontFamily="body"
          fontSize={
            Array [
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          id="home-office"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/products?prop=home-office"
            onClick={[Function]}
          >
            Ruko
          </a>
        </li>
      </ul>
      <div
        className="sc-AxmLO fuxBiu"
        display="flex"
        width={
          Array [
            0,
            0,
            "220px",
          ]
        }
      >
        <svg
          fill="none"
          height={220}
          id="house-svg"
          width={220}
        >
          <path
            d="M179.682 57.416L160.773 41.27V26.8h18.909v30.616zM22.265 214.586V100.742l88.237-69.607 88.104 68.61v114.841H22.265z"
            stroke="#fff"
          />
          <path
            d="M129.532 74.73c0 11.002-8.606 19.89-19.186 19.89S91.16 85.732 91.16 74.73c0-11.001 8.606-19.89 19.186-19.89s19.186 8.889 19.186 19.89zM147.053 149.597v-35.044h33.8v35.044h-33.8z"
            stroke="#fff"
          />
          <path
            d="M219.47 89.978l-2.895 23.354-105.726-82.63-.31-.244-.309.246L5.063 114.4.543 88.215 110.554.642 219.47 89.978z"
            stroke="#fff"
          />
          <path
            d="M217.77 91.174l-3.067 17.721-103.872-84.786-.315-.257-.316.256L5.278 108.909 2.23 91.169l108.3-87.53 107.24 87.535zM89.405 214.586v-59.257h41.534v59.257H89.405z"
            stroke="#fff"
          />
          <path
            d="M90.633 214.586v-57.982h39.075v57.982H90.633zM148.285 126.659v-10.834h31.34v10.834h-31.34z"
            stroke="#fff"
          />
          <path
            d="M148.285 148.321v-20.662h31.34v20.662h-31.34z"
            opacity={0.2}
            stroke="#fff"
          />
          <path
            d="M116.838 91.243a16.891 16.891 0 01-6.667 1.374c-2.288 0-4.554-.467-6.668-1.374a17.405 17.405 0 01-5.656-3.918 18.126 18.126 0 01-3.78-5.868 18.638 18.638 0 01-1.318-6.426h34.843a18.614 18.614 0 01-1.318 6.426 18.129 18.129 0 01-3.78 5.868 17.413 17.413 0 01-5.656 3.918z"
            stroke="#fff"
          />
          <path
            d="M122.488 87.15c-3.27 3.386-7.701 5.286-12.317 5.286-4.617 0-9.048-1.9-12.318-5.286-3.156-3.27-4.98-7.667-5.104-12.283h34.843c-.123 4.616-1.947 9.013-5.104 12.283zM97.859 61.76c3.269-3.384 7.698-5.28 12.312-5.28 4.613 0 9.042 1.896 12.311 5.28 3.156 3.265 4.982 7.658 5.11 12.271H92.75c.127-4.613 1.953-9.006 5.109-12.272z"
            opacity={0.2}
            stroke="#fff"
          />
          <path
            d="M97.859 61.76c3.269-3.384 7.698-5.28 12.312-5.28 4.613 0 9.042 1.896 12.311 5.28 3.156 3.265 4.982 7.658 5.11 12.271H92.75c.127-4.613 1.953-9.006 5.109-12.272z"
            opacity={0.2}
            stroke="#fff"
          />
          <path
            d="M169.74 188.365l.535-.294-.394-.467a11.468 11.468 0 01-2.248-4.219 11.75 11.75 0 01-.357-4.81 11.625 11.625 0 011.601-4.528 11.164 11.164 0 013.264-3.426l.4-.27-.256-.409a6.67 6.67 0 01-1.007-3.547h0v-.014a6.404 6.404 0 011.42-4.2 5.966 5.966 0 013.734-2.156 5.847 5.847 0 014.168.955 6.255 6.255 0 012.517 3.597l.313 1.197.606-1.078a9.377 9.377 0 013.355-3.462 8.947 8.947 0 014.559-1.274c5.095 0 9.247 4.372 9.247 9.827v.001a10.34 10.34 0 01-.569 3.401l-.23.663h.702c.119 0 .235-.009.329-.016l.008-.001c.102-.008.182-.014.26-.014 4.02 0 7.326 3.46 7.326 7.79 0 .29-.001.532-.035.762l-.055.369.339.158a9.56 9.56 0 014.081 3.635 10.076 10.076 0 011.507 5.366v.003c0 5.437-4.152 9.792-9.217 9.793a8.818 8.818 0 01-1.596-.149l-.647-.121.057.656c.029.336.045.671.045 1.005 0 6.13-4.685 11.051-10.406 11.051h0a9.938 9.938 0 01-5.912-1.961l-.329-.244-.304.276a7.348 7.348 0 01-4.935 1.929c-4.195-.001-7.645-3.611-7.645-8.129 0-.213.001-.378.025-.538l.086-.574h-.611c-4.516 0-8.223-3.887-8.223-8.741v-.002a9.021 9.021 0 011.2-4.549 8.58 8.58 0 013.292-3.241z"
            stroke="#fff"
          />
          <path
            d="M182.856 194.87l-.446.49.593.294a7.182 7.182 0 012.915 2.678 7.557 7.557 0 011.114 3.878 7.384 7.384 0 01-7.38 7.382h-.612l.091.578c.02.126.021.261.021.442 0 3.789-3.082 6.861-6.856 6.861h-.002a6.818 6.818 0 01-4.436-1.629l-.292-.248-.315.218a9.292 9.292 0 01-5.322 1.659h-.002c-5.16 0-9.347-4.19-9.347-9.356 0-.285.013-.564.039-.842l.06-.651-.645.113a8.284 8.284 0 01-1.432.125c-4.568 0-8.272-3.706-8.273-8.281a8.505 8.505 0 011.39-4.594 8.04 8.04 0 013.61-3.013l.343-.143-.038-.369a6.567 6.567 0 016.532-7.219c.068 0 .139.006.232.013h.008c.085.007.191.015.3.015h.716l-.247-.672a8.286 8.286 0 01-.509-2.868v0c0-4.59 3.717-8.311 8.303-8.311h.004a8.126 8.126 0 014.05 1.056 8.5 8.5 0 013.059 2.962l.58.946.324-1.062a5.59 5.59 0 011.978-2.843 5.24 5.24 0 013.207-1.059h.006a5.411 5.411 0 014.513 8.39l-.287.432.442.271a9.235 9.235 0 013.242 3.405 9.664 9.664 0 011.203 4.624 9.431 9.431 0 01-2.434 6.328z"
            stroke="#fff"
          />
        </svg>
        <svg
          fill="none"
          height={220}
          id="apartment-svg"
          width={221}
        >
          <path
            d="M204.5 205.75V19.824h15V205.75h-15z"
            stroke="#fff"
          />
          <path
            d="M220 19.324V206.25h-16"
            opacity={0.1}
            stroke="#fff"
          />
          <path
            d="M.5 218.5v-161h58v161H.5zM58.518 219.5V.5h145.628v219H58.518z"
            stroke="#fff"
          />
          <path
            d="M16.465 93.149V67.392h29.986V93.15H16.465zM16.465 138.486v-25.757h29.986v25.757H16.465zM16.465 183.824v-25.757h29.986v25.757H16.465zM77.33 47.811V22.054h29.985v25.757H77.33zM77.33 93.149V67.392h29.985V93.15H77.33zM77.33 138.486v-25.757h29.985v25.757H77.33zM185.333 22.054v25.757h-29.986V22.054h29.986zM185.333 67.392v25.756h-29.986V67.392h29.986zM185.333 112.73v25.756h-29.986V112.73h29.986zM118.275 219.5v-70.352h26.112V219.5h-26.112z"
            stroke="#fff"
          />
          <path
            d="M77.33 29.23v-7.176h29.985v7.176H77.33zM155.347 29.23v-7.176h29.986v7.176h-29.986zM155.347 74.568v-7.176h29.986v7.176h-29.986zM155.347 119.905v-7.176h29.986v7.176h-29.986zM77.33 74.568v-7.176h29.985v7.176H77.33zM77.33 119.905v-7.176h29.985v7.176H77.33zM16.465 72.71v-5.318h29.986v5.318H16.465zM16.465 118.047v-5.318h29.986v5.318H16.465zM16.465 163.385v-5.318h29.986v5.318H16.465z"
            opacity={0.1}
            stroke="#fff"
          />
        </svg>
        <svg
          fill="none"
          height={153}
          id="kavling-svg"
          width={228}
        >
          <g
            filter="url(#kavling-hover_svg__filter0_d)"
          >
            <path
              d="M110.205 133.966l.253.098.22-.159 100.919-72.858 11.262 3.153-112.547 80.238L5.089 104.875l10.973-7.315 94.143 36.406z"
              stroke="#fff"
            />
          </g>
          <g
            filter="url(#kavling-hover_svg__filter1_d)"
          >
            <path
              d="M5.103 93.88l105.372-68.343 112.348 27.688-112.511 80.213L5.103 93.881z"
              stroke="#fff"
            />
          </g>
          <g
            filter="url(#kavling-hover_svg__filter2_d)"
            stroke="#fff"
          >
            <path
              d="M113.318 4.2l7.327 8.4 8.29-8.4"
              strokeMiterlimit={10}
            />
            <path
              d="M106.877 32.5V.5h3.628v32h-3.628z"
            />
            <path
              d="M107.456 31.3V1.1h3.049v30.2h-3.049z"
            />
            <path
              d="M105.721 6.7v-5h27.92v5h-27.92z"
            />
            <path
              d="M133.063 6.1h-26.764V2.3h26.764v3.8zM112.662 25.3V12.5h16.93v12.8h-16.93z"
            />
            <path
              d="M112.662 24.7V12.5h16.93v12.2h-16.93z"
            />
            <path
              d="M114.975 15.1v-.2H126.7v.2h-11.725zM117.289 17.5v-.2h7.676v.2h-7.676zM116.132 19.9v-.2h9.99v.2h-9.99zM114.975 22.3v-.2H126.7v.2h-11.725z"
            />
          </g>
          <defs>
            <filter
              colorInterpolationFilters="sRGB"
              filterUnits="userSpaceOnUse"
              height={92.5}
              id="kavling-hover_svg__filter0_d"
              width={228}
              x={0}
              y={60.5}
            >
              <feFlood
                floodOpacity={0}
                result="BackgroundImageFix"
              />
              <feColorMatrix
                in="SourceAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feOffset
                dy={4}
              />
              <feGaussianBlur
                stdDeviation={2}
              />
              <feColorMatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feBlend
                in2="BackgroundImageFix"
                result="effect1_dropShadow"
              />
              <feBlend
                in="SourceGraphic"
                in2="effect1_dropShadow"
                result="shape"
              />
            </filter>
            <filter
              colorInterpolationFilters="sRGB"
              filterUnits="userSpaceOnUse"
              height={117}
              id="kavling-hover_svg__filter1_d"
              width={228}
              x={0}
              y={25}
            >
              <feFlood
                floodOpacity={0}
                result="BackgroundImageFix"
              />
              <feColorMatrix
                in="SourceAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feOffset
                dy={4}
              />
              <feGaussianBlur
                stdDeviation={2}
              />
              <feColorMatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feBlend
                in2="BackgroundImageFix"
                result="effect1_dropShadow"
              />
              <feBlend
                in="SourceGraphic"
                in2="effect1_dropShadow"
                result="shape"
              />
            </filter>
            <filter
              colorInterpolationFilters="sRGB"
              filterUnits="userSpaceOnUse"
              height={41}
              id="kavling-hover_svg__filter2_d"
              width={36.92}
              x={101.221}
              y={0}
            >
              <feFlood
                floodOpacity={0}
                result="BackgroundImageFix"
              />
              <feColorMatrix
                in="SourceAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feOffset
                dy={4}
              />
              <feGaussianBlur
                stdDeviation={2}
              />
              <feColorMatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feBlend
                in2="BackgroundImageFix"
                result="effect1_dropShadow"
              />
              <feBlend
                in="SourceGraphic"
                in2="effect1_dropShadow"
                result="shape"
              />
            </filter>
          </defs>
        </svg>
        <svg
          fill="none"
          height={220}
          id="home-office-svg"
          width={220}
        >
          <path
            d="M.5 36.5v-.528L50.241.5h125.505l-40.437 35.503-33.809.497H.5zM199.726 20.034v.228l.173.15 19.601 16.96V219.5h-83.065V35.942L177.073.662l5.965 5.161.827.716V1.213h15.861v18.82z"
            stroke="#fff"
          />
          <path
            d="M.5 219.5V36.214h134.935V219.5H.5z"
            stroke="#fff"
          />
          <path
            d="M53.264 219.5v-44h24.04v44h-24.04zM15.703 181.214h9.732v6.143h-9.732v-6.143zM15.703 200.214v-6.143h9.732v6.143h-9.732zM33.59 187.357v-6.143h9.731v6.143H33.59zM33.59 200.214v-6.143h9.731v6.143H33.59zM89.037 181.214h9.731v6.143h-9.731v-6.143zM89.037 200.214v-6.143h9.731v6.143h-9.731zM106.923 187.357v-6.143h9.732v6.143h-9.732zM106.923 200.214v-6.143h9.732v6.143h-9.732zM164.158 38.357h9.732V44.5h-9.732v-6.143zM164.158 57.357v-6.143h9.732v6.143h-9.732zM182.045 44.5v-6.143h9.732V44.5h-9.732zM182.045 57.357v-6.143h9.732v6.143h-9.732zM164.158 78.357h9.732V84.5h-9.732v-6.143zM164.158 97.357v-6.143h9.732v6.143h-9.732zM182.045 84.5v-6.143h9.732V84.5h-9.732zM182.045 97.357v-6.143h9.732v6.143h-9.732zM164.158 118.357h9.732v6.143h-9.732v-6.143zM164.158 137.357v-6.143h9.732v6.143h-9.732zM182.045 124.5v-6.143h9.732v6.143h-9.732zM182.045 137.357v-6.143h9.732v6.143h-9.732zM164.158 158.357h9.732v6.143h-9.732v-6.143zM164.158 177.357v-6.143h9.732v6.143h-9.732zM182.045 164.5v-6.143h9.732v6.143h-9.732zM182.045 177.357v-6.143h9.732v6.143h-9.732zM15.703 61.643V45.5h102.74v16.143H15.703zM15.703 95.928V79.787h102.74v16.142H15.703zM15.703 130.214v-16.143h102.74v16.143H15.703zM15.703 164.5v-16.143h102.74V164.5H15.703z"
            stroke="#fff"
          />
        </svg>
      </div>
    </div>
  </li>
  <button
    className="sc-fzoLsD cmsPyc"
    display={
      Array [
        "block",
        "block",
        "none",
      ]
    }
    fontFamily="body"
    fontSize={
      Array [
        1,
        1,
        2,
        2,
      ]
    }
    fontWeight="bold"
    onClick={[MockFunction]}
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M16.6 14c-.2-.1-1.5-.7-1.7-.8c-.2-.1-.4-.1-.6.1c-.2.2-.6.8-.8 1c-.1.2-.3.2-.5.1c-.7-.3-1.4-.7-2-1.2c-.5-.5-1-1.1-1.4-1.7c-.1-.2 0-.4.1-.5c.1-.1.2-.3.4-.4c.1-.1.2-.3.2-.4c.1-.1.1-.3 0-.4c-.1-.1-.6-1.3-.8-1.8c-.1-.7-.3-.7-.5-.7h-.5c-.2 0-.5.2-.6.3c-.6.6-.9 1.3-.9 2.1c.1.9.4 1.8 1 2.6c1.1 1.6 2.5 2.9 4.2 3.7c.5.2.9.4 1.4.5c.5.2 1 .2 1.6.1c.7-.1 1.3-.6 1.7-1.2c.2-.4.2-.8.1-1.2l-.4-.2m2.5-9.1C15.2 1 8.9 1 5 4.9c-3.2 3.2-3.8 8.1-1.6 12L2 22l5.3-1.4c1.5.8 3.1 1.2 4.7 1.2c5.5 0 9.9-4.4 9.9-9.9c.1-2.6-1-5.1-2.8-7m-2.7 14c-1.3.8-2.8 1.3-4.4 1.3c-1.5 0-2.9-.4-4.2-1.1l-.3-.2l-3.1.8l.8-3l-.2-.3c-2.4-4-1.2-9 2.7-11.5S16.6 3.7 19 7.5c2.4 3.9 1.3 9-2.6 11.4\\" fill=\\"#F2D9C6\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
    Hubungi Saya!
  </button>
</ul>
`;
